### PR TITLE
common/thanos: revise thanos alerts

### PR DESCRIPTION
**Pull request by richardtief**

### CODE REVIEW

File: common/thanos/templates/alerts/_thanos-compactor.alerts.tpl [hunk 1]
```diff
@@ -3,83 +3,109 @@
 groups:
 - name: thanos-compactor.alerts
   rules:
-    - alert: ThanosCompactHalted
-      expr: thanos_compact_halted{app="thanos-compactor", prometheus="{{ include "thanos.name" . }}"} == 1
+    - alert: ThanosCompactMultipleRunning
+      expr: sum by (thanos) (up{job=~".*thanos.*compact.*", thanos="{{ include "thanos.name" . }}"}) > 1
```
Comment (viennaa) [inline common/thanos/templates/alerts/_thanos-compactor.alerts.tpl:7]: Why don't we sum by `job` here?

File: common/thanos/templates/alerts/_thanos-compactor.alerts.tpl [hunk 2]
```diff
@@ -3,83 +3,109 @@
...
+    - alert: ThanosCompactHalted
+      expr: thanos_compact_halted{job=~".*thanos.*compact.*", thanos="{{ include "thanos.name" . }}"} == 1
+      for: 5m
       labels:
-        context: thanos
         service: {{ default "metrics" $root.Values.alerts.service }}
-        support_group: observability
+        support_group: {{ default "observability" $root.Values.alerts.support_group }}
         severity: info
         playbook: 'docs/support/playbook/prometheus/thanos_compaction.html'
```
Comment (viennaa) [inline common/thanos/templates/alerts/_thanos-compactor.alerts.tpl:28]: FWIW: the playbook does not exist

File: common/thanos/templates/alerts/_thanos-store.alerts.tpl [hunk 3]
```diff
@@ -4,29 +4,82 @@ groups:
...
+      labels:
+        service: {{ default "metrics" $root.Values.alerts.service }}
+        support_group: {{ default "observability" $root.Values.alerts.support_group }}
+        severity: info
+        meta: Thanos Store `{{`{{ $labels.thanos }}`}}` has a 99th percentile latency for store series gate requests.
+      annotations:
+        description: |
+          Thanos Store `{{`{{ $labels.thanos }}`}}` has a 99th 
+          percentile latency of `{{`{{ $value }}`}}`seconds 
+          for store series gate requests.
```
Comment (viennaa) [inline common/thanos/templates/alerts/_thanos-store.alerts.tpl:43]: Would be interesting to know what `series gate requests` are about. Guess we have to figure this out, once it happens. Hence `info` is good 👍

**Review by viennaa — COMMENTED** [review]
Thanks again for also bringing these alerts back on track. Why did you swap `job` with `thanos` in the queries. The job itself would give a little bit more insight, wouldn't it? I noticed this in `prometheus-server` already but didn't commented it there.

File: common/thanos/templates/alerts/_thanos-compactor.alerts.tpl (hunk 1 above)
Comment (richardtief) [reply common/thanos/templates/alerts/_thanos-compactor.alerts.tpl:7]: sum by `job` would mean that `$labels.thanos` is not resolved in the meta.

File: common/thanos/templates/alerts/_thanos-compactor.alerts.tpl (hunk 2 above)
Comment (richardtief) [reply common/thanos/templates/alerts/_thanos-compactor.alerts.tpl:28]: Yeah, I know. Let's create the playbooks step by step afterwards when we need to respond to one of the new alerts. Monitoring the #cc-metrics channel after deployment.
```suggestion
```

**Comment by richardtief** [comment]
Since we tag the job's metrics with the Thanos name, we can use the `thanos` label to refer to the affected resource. 

https://github.com/sapcc/helm-charts/blob/master/common/thanos/templates/servicemonitor.yaml#L27-L29

**Review by viennaa — APPROVED** [review]
Good to go from my side.

**Pull request merged**